spec/queries/target_query_spec.rb

```diff
@@ -0,0 +1,100 @@
+require 'rails_helper'
+
+describe TargetQuery do
+  describe '#compatible_target' do
+    subject do
+      described_class.new.compatible_target(new_target)
+    end
+    let!(:user) { create(:user) }
+    let!(:topic) { create(:topic) }
+    let(:topic_id) { topic.id }
+
+    let(:new_target) do
+      build(:target,
+            radius: 25,
+            latitude: 60,
+            longitude: -60,
+            topic_id: topic_id,
+            user: user)
+    end
+
+    context 'when there are targets in range' do
+      context 'when there are targets with the same topic' do
+        context 'when there are targets without a match' do
+          context 'when there are targets with different users' do
+            let!(:target) do
+              create(:target,
+                     radius: 20,
+                     latitude: 60,
+                     longitude: -60,
+                     topic_id: topic_id,
+                     matched: false)
+            end
+
+            it 'returns a complatible target' do
+              expect(subject).to eq(target)
+            end
+          end
+          context 'when there are targers with same user' do
```

Comment:
```suggestion
          end
          
          context 'when there are targers with same user' do
```

use empty lines to slit between `it's` and `context's` or generally to split ideas